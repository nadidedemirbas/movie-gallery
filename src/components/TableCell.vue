<template>
    <div class="index-cell">
      <router-link :to="path" >
        <div>
          <div class="index-image-wrapper">
            <img class="image-fluid" :src="image" :alt="title">
          </div>
          <div class="index-movie-title-wrapper">
            {{ title }}
          </div>
        </div>
      </router-link>
    </div>
</template>

<script>
export default {
  name: "TableCell",
  props: {
    image: String,
    title: String,
    path: { type: [String, Object], default: "" }
  },
}
</script>

<style scoped>

</style>