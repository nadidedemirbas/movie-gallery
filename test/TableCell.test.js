import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'

import TableCell from '../src/components/TableCell.vue'

describe('TableCell Component', () => {
    const wrapper = mount(TableCell, {
        props: {
            image: "mockimage",
            title: "mock-title",
            path: "mockpath"
        }
    })

    it('should render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    it('should show title correctly', () => {
        const title = wrapper.find('.index-movie-title-wrapper').text()
        expect(title).toBe('mock-title');
    })

    it('should show poster correctly', () => {
        const attributes = wrapper.find('.image-fluid').attributes();
        const source = attributes.src;
        expect(source).toBe('mockimage');
    })
});