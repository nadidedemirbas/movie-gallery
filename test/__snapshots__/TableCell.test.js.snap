// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableCell Component > should render correctly 1`] = `
"<div class=\\"index-cell\\">
  <router-link to=\\"mockpath\\">
    <div>
      <div class=\\"index-image-wrapper\\"><img class=\\"image-fluid\\" src=\\"mockimage\\" alt=\\"mock-title\\"></div>
      <div class=\\"index-movie-title-wrapper\\">mock-title</div>
    </div>
  </router-link>
</div>"
`;
